packages/UI/src/components/progressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,66 @@
+import * as Progress from '@radix-ui/react-progress';
+import React from 'react';
+import { tv } from 'tailwind-variants';
+
+interface ProgressProps {
+  startColor?: string;
+  endColor?: string;
+  backgroundColor?: string;
+  width?: string;
+  height?: string;
+}
+
+const ProgressBar: React.FC<ProgressProps> = ({
+  startColor = '#394CAC',
+  endColor = '#7339AC',
+  backgroundColor = '#F0EDEE',
+  width = '332px',
+  height = '25px',
+}) => {
+  const [progress, setProgress] = React.useState(0);
+
+  React.useEffect(() => {
+    const timer = setInterval(() => {
+      setProgress((currentProgress) => {
+        const newProgress = currentProgress + 5;
+        if (newProgress >= 100) {
+          clearInterval(timer);
+          return 100;
+        }
+        return newProgress;
+      });
+    }, 500);
+
+    return () => clearInterval(timer);
+  }, []);
+  const currentColor = progress >= 95 ? endColor : startColor;
+  const indicatorStyle = {
+    height: '100%',
+    backgroundColor: `${currentColor}`,
+    width: `${progress}%`,
+  };
+
+  // const progressClass = progress === 100 ? endColor : startColor;
+
+  return (
+    <>
+      <Progress.Root
+        className="rounded-2xl relative overflow-hidden"
+        style={{
+          width,
+          height,
+          backgroundColor,
+        }}
```

Comment:
> Yeah I tried that, but since the colors are dynamic in the progress bar it didn't work as I expected it to be, but I'll try to find a solution.

Read this and you'll have an idea of how to do it
https://tailwindcss.com/docs/content-configuration#dynamic-class-names